crates/starknet/src/starknet/mod.rs

```diff
@@ -274,13 +275,24 @@ impl Starknet {
                 // based on this https://community.starknet.io/t/efficient-utilization-of-sequencer-capacity-in-starknet-v0-12-1/95607#the-validation-phase-in-the-gateway-5
                 // we should not save transactions that failed with one of the following errors
                 match tx_err {
-                    blockifier::transaction::errors::TransactionExecutionError::InvalidNonce { .. } =>
+                    blockifier::transaction::errors::TransactionExecutionError::TransactionPreValidationError(TransactionPreValidationError::InvalidNonce { .. }) =>
                         Err(TransactionValidationError::InvalidTransactionNonce.into()),
-                    blockifier::transaction::errors::TransactionExecutionError::MaxFeeExceedsBalance { .. } =>
-                        Err(TransactionValidationError::InsufficientAccountBalance.into()),
-                    blockifier::transaction::errors::TransactionExecutionError::FeeTransferError { .. }
-                    | blockifier::transaction::errors::TransactionExecutionError::MaxFeeTooLow { .. } =>
+                    blockifier::transaction::errors::TransactionExecutionError::FeeCheckError { .. } =>
                         Err(TransactionValidationError::InsufficientMaxFee.into()),
+                    blockifier::transaction::errors::TransactionExecutionError::TransactionPreValidationError(
+                        TransactionPreValidationError::TransactionFeeError(tx_fee_err)
+                    ) => match tx_fee_err {
+                        blockifier::transaction::errors::TransactionFeeError::FeeTransferError { .. }
+                        | blockifier::transaction::errors::TransactionFeeError::MaxFeeTooLow { .. } => Err(
+                            TransactionValidationError::InsufficientMaxFee.into()
+                        ),
+                        blockifier::transaction::errors::TransactionFeeError::MaxFeeExceedsBalance { .. } => Err(
+                            TransactionValidationError::InsufficientAccountBalance.into()
+                        ),
+                        _ => Err(tx_fee_err.into())
+                    },
+                    // blockifier::transaction::errors::TransactionExecutionError::TransactionFeeError { .. } =>
```

Comment:
what about TransactionFeeError now its going to fall in the default case

Reply:
As discussed on Discord, I addressed this in https://github.com/0xSpaceShard/starknet-devnet-rs/pull/256/commits/87de6234ab06c098cc7859b5a97f61502907ef00